### PR TITLE
Add appcatalogentry resource with delete support

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#10930

Replaces the empty resource with the new resource for the appcatalog controller.

This PR only implements delete to make it easier to review.

### CODE REVIEW

File: service/controller/appcatalog/resources.go [hunk 1]
```diff
@@ -24,13 +29,23 @@ func newAppCatalogResources(config appCatalogResourcesConfig) ([]resource.Interf
...
+
+		appCatalogEntryResource, err = appcatalogentry.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
 	}
 
 	resources := []resource.Interface{
-		emptyResource,
+		appCatalogEntryResource,
```
Comment (rossf7) [inline service/controller/appcatalog/resources.go:48]: The empty resource gets replaced.

**Comment by rossf7** [comment]
Tested on `ghost`.

File: service/controller/appcatalog/resource/appcatalogentry/resource.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	entries, err := r.k8sClient.G8sClient().ApplicationV1alpha1().AppCatalogEntries(metav1.NamespaceDefault).List(ctx, lo)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	for _, entry := range entries.Items {
+		currentEntryCRs[entry.Name] = entry.DeepCopy()
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("got %d appcatalogentries for appcatalog %#q", len(currentEntryCRs), cr.Name))
```
Comment (tomahawk28) [inline service/controller/appcatalog/resource/appcatalogentry/resource.go:76]: Is that something we can print out in `EnsureDeleted`?

**Review by tomahawk28 — APPROVED** [review]
Just a one comment LGTM

File: service/controller/appcatalog/resource/appcatalogentry/resource.go (hunk 2 above)
Comment (rossf7) [reply service/controller/appcatalog/resource/appcatalogentry/resource.go]: Good point. I've changed it to `getting current appcatalogentries` and `got x current appcatalogentries`.

There is also logs for deleting the CRs so it should be clear now.

**Pull request merged**